dev/source/docs/setting-up-sitl-on-macosx.rst

```diff
@@ -0,0 +1,137 @@
+Setting up SITL on Mac OS X
+===========================
+
+This page describes how to setup the :ref:`SITL (Software In The Loop) <sitl-simulator-software-in-the-loop>` on Mac OS X. The specific commands were tested on macOS Catalina 10.15.7 and macOS Big Sur 11.1.
+
+Overview
+========
+
+The SITL simulator allows you to run Plane, Copter or Rover without any
+hardware. It is a build of the autopilot code using an ordinary C++
+compiler, giving you a native executable that allows you to test the
+behaviour of the code without hardware.
+
+SITL runs natively on Linux, Windows and Mac OSX.
+
+.. image:: ../images/SITL_macosx.png
+    :target: ../_images/SITL_macosx.png
+
+
+Install steps
+=============
+
+This guide assumes that a build environment is already setup for Ardupilot. If that is not the case please follow the available guides to setup the build environment for Mac OSX. The following step provide a condensed versions of the steps to take.
+
+
+Prerequisites
+-------------
+
+The following tools need to be installed to complete the installation. Between brackets the version used by the author at time of writing.
+
+ * Xcode (12.4)
+ * Xcode commandline tools (12.4)
+ * Homebrew (3.0.4-52-gbc37074)
+
+Preparing Python
+----------------
+
+Install the required python 3 version using homebrew and activate it. If you are familiar with pyenv or virtualenv it might be better to use one of those to setup the environment. This limits potential problems with other tools that require specific versions of python. How to do that is beyond the scope of this document.
+
+The following steps install python version 3.9.x for system wide usage.
+
+::
+
+	brew install python@3.9
+  export PATH="$(brew --prefix)/opt/python/libexec/bin":$PATH
+	
+	pip install wheel
+	pip install future pymavlink mavproxy
+
+Optional: Add the following to the either .bashrc or .zshrc to enable this version of python be default
+
+::
+
+  export PATH="$(brew --prefix)/opt/python/libexec/bin":$PATH
+
+
+Optional: Install sfml (used by sitl for audio and osd)
+
+::
+
+  brew install sfml
+
+Note that not all sfml features work on Mac OSX currently. The following error can appear `Cannot create a window from a worker thread. (OS X limitation)`, this is a result of Mac OSX specifics that haven't been catered for in the sitl code yet.
+
+Start SITL simulator
+--------------------
+
+.. warning::
+
+	Due to ideosyncracies with Python and using screens there are limitations when Python needs to use the screen to display a graphical UI. Python 3.9 solves part of the problem by no longer requiring a separate pythonw binary. However the requirement to run on the main display still stands. When observing the error "This program needs access to the screen. Please run with a Framework build of python, and only when you are logged in on the main display of your Mac.", disconnect external displays during startup. When SITL is started the windows can be moved around as normal.
```

Comment:
sp.

```diff
- ideosyncracies
+ idiosyncrasies
```

Reply:
I haven't been able to get this to work at all on my M1 Mac. So I just run SiTL with no UI. It works fine and I connect to with with QGroundControl which shows the map and all the messages.
Maybe the message should say the UI is optional?

Reply:
@timtuxworth unfortunately I don't have a M1 to test at present - still Intel based machines for me. When I get round to upgrading my macbook I'll take another look at the UI modules in MAVProxy which seem to be causing the problems. As you suggest it might be worth emphasising the difference between a bare bones SITL install (with alternative GCS) and full MAVProxy support for macOS which should have all the UI elements functioning correctly.

Reply:
Indeed; it seems like most of the difficulties that arise in installing SITL stuff on an M1 Mac are due to the GUI stuff. I run the SITL without a GUI and I haven't encountered any significant issues.